content/posts/2025-01-28-cli_use_cases.md

```diff
@@ -0,0 +1,365 @@
+```
+
+Similarly to the previous scenario, the script will connect to your cluster and start deploying eBPF agents and collector pod but capturing full packet content this time.
+
+Once first packet is captured, you will see a table with the amount of packets and bytes:
+
+![cli packets enriched]({page.image('cli/packet-capture-scenario-table-enriched.png')})
+
+You can cycle between the view to see IPs and ports instead of resources:
+
+![cli packets ips]({page.image('cli/packet-capture-scenario-table-ips.png')})
+
+Let the capture run enough time to get some packets and press `CTRL + C` to exit.
+Open the generated pcap file using Wireshark and look for your IP as source for example:
+
+![wireshark unencrypted packets]({page.image('cli/wireshark-unencrypted-pcap.png')})
+
+This time, we can see the HTTP content and see the request URL and response. Since we have the entire payload, we cand check for any authentification header, cookie, encoding and so on that may be wrong. 
+This is useful to identify traffic content but also to debug a deployment that may be misconfigured or requesting unexpected content on unencrypted connections.
+
+## Encrypted traffic
+
+What about encrypted traffic ? 
+
+Most of the time, your traffic will be secured and you will not have access to the data it holds. However, Wireshark has the ability to decrypt the content if you [provide the proper secrets](https://wiki.wireshark.org/TLS#tls-decryption). However, getting these secrets may be tricky depending on where the encryption happens and needs some juggling. 
+If you are looking for the traffic between your browser and the cluster, you can simply run the following command to capture [pre-master-secret keylog file](https://wiki.wireshark.org/TLS#using-the-pre-master-secret):
+```sh
+SSLKEYLOGFILE="keylog.txt" chromium-browser &
+```
+
+This generated file contains each client and server traffic secrets to decode your packets.
+
+Then in Wireshark, specify the file path in `Edit→Preferences→Protocols→TLS→(Pre)-Master-Secret log filename`. A single file can hold keys for multiple TLS connections.
+
+![wireshark tls keylog]({page.image('cli/wireshark-tls-keylog.png')})
+
+You should now have access to decrypted packets !
+
+What about a secure connection between two pods ? 
+For these cases, you need to have a good understanding of what's involved behind the scene. If your application rely on OpenSSL, GnuTLS or NSS TLS library, you should be able to set the `SSLKEYLOGFILE` enviroment variable and extract the result.
+
+For our nodejs example, you need to either use the cli option [--tls-keylog](https://nodejs.org/api/cli.html#cli_tls_keylog_file) or modify the code and subscribe on [keylog event](https://nodejs.org/api/tls.html#event-keylog). When multiple teams are involved, that may take a while !
+
+Another alternative is to get a man in the middle approach using a proxy. This can be done using [kubetap](https://soluble-ai.github.io/kubetap/).
```

Comment:
```suggestion
Another alternative is a man-in-the-middle approach using a proxy. This can be done using [kubetap](https://soluble-ai.github.io/kubetap/).
```